File: backend/src/main/java/com/back/domain/notification/dto/response/NotificationResponseDto.java
package com.back.domain.notification.dto.response;

import com.back.domain.notification.entity.Notification;
import com.back.domain.notification.enums.NotificationType;
import lombok.Builder;

import java.time.LocalDateTime;

@Builder
public record NotificationResponseDto (
        Long notificationId,
        NotificationType type,
        String title,
        String message,
        boolean isRead,
        LocalDateTime createdAt
){
    public static NotificationResponseDto from(Notification notification) {

        return NotificationResponseDto.builder()
                .notificationId(notification.getId())
                .type(notification.getType())
                .title(notification.getTitle())
                .message(notification.getMessage())
                .isRead(notification.isRead())
                .createdAt(notification.getCreatedAt())
                .build();
    }

}

File: backend/src/main/java/com/back/domain/notification/dto/response/NotificationResponseDto.kt
package com.back.domain.notification.dto.response

import com.back.domain.notification.entity.Notification
import com.back.domain.notification.enums.NotificationType
import java.time.LocalDateTime

data class NotificationResponseDto(
    val notificationId: Long?,
    val type: NotificationType,
    val title: String,
    val message: String,
    val isRead: Boolean,
    val createdAt: LocalDateTime?
) {
    companion object {
        fun from(notification: Notification): NotificationResponseDto {
            return NotificationResponseDto(
                notificationId = notification.id,
                type = notification.type,
                title = notification.title,
                message = notification.message,
                isRead = notification.isRead,
                createdAt = notification.createdAt
            )
        }
    }
}
